ReportPortal.Shared.Tests/ConfigurationTest.cs

```diff
@@ -0,0 +1,198 @@
+
+            var config = new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build();
+
+            Assert.Equal(1, config.Values.Count);
+
+            var variable = config.GetValue<string>("prop1__prop2");
+            Assert.Equal("value2", variable);
+        }
+
+        [Fact]
+        public void ShouldGetListFromJsonFile()
+        {
+            var tempFile = Path.GetTempFileName();
+
+            File.WriteAllText(tempFile, "{\"prop1\": [\"value1\", \"value2\"]}");
+
+            var config = new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build();
+
+            var variable = config.GetValue<string>("prop1");
+            Assert.Equal("value1;value2;", variable);
+
+            var list = config.GetValues<string>("prop1");
+            Assert.Equal(new List<string> { "value1", "value2" }, list);
+        }
+
+        [Fact]
+        public void ShouldGetListOfStringsFromJsonFile()
+        {
+            var tempFile = Path.GetTempFileName();
+
+            File.WriteAllText(tempFile, "{\"prop1\": [\"value1\", \"value2\"]}");
+
+            var config = new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build();
+
+            var list = config.GetValues<string>("prop1");
+            Assert.Equal(new List<string> { "value1", "value2" }, list);
+        }
+
+        [Fact]
+        public void ShouldGetListOfIntegersFromJsonFile()
+        {
+            var tempFile = Path.GetTempFileName();
+
+            File.WriteAllText(tempFile, "{\"prop1\": [1, 2]}");
+
+            var config = new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build();
+
+            var list = config.GetValues<int>("prop1");
+            Assert.Equal(new List<int> { 1, 2 }, list);
+        }
+
+        [Fact]
+        public void ShouldGetSeveralListsFromJsonFile()
+        {
+            var tempFile = Path.GetTempFileName();
+
+            File.WriteAllText(tempFile, "{\"prop1\": [\"value1\", \"value2\"], \"prop2\": [\"value11\", \"value22\"]}");
+
+            var config = new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build();
+
+            Assert.Equal(2, config.Values.Count);
+
+            var variable1 = config.GetValue<string>("prop1");
+            Assert.Equal("value1;value2;", variable1);
+
+            var variable2 = config.GetValue<string>("prop2");
+            Assert.Equal("value11;value22;", variable2);
+        }
+
+        [Fact]
+        public void ShouldRaiseExceptionIfJsonIsIncorrect()
+        {
+            var tempFile = Path.GetTempFileName();
+
+            File.WriteAllText(tempFile, "Bh}");
+
+            Assert.Throws<XmlException>(() => new ConfigurationBuilder().AddJsonFile(filePath: tempFile).Build());
+        }
+
+        [Fact]
+        public void ShouldRaiseExceptionIfVariableNotFound()
+        {
+            var config = new ConfigurationBuilder().Build();
+
+            Assert.Throws<KeyNotFoundException>(() => config.GetValue<string>("a"));
```

Comment:
It might be useful to have an overload of GetValue() that accepts a default value and doesn't throw an exception if key is not found.